### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by ColdIce1605**
Cleans up the ProgramBuilder class ie removes heavily duplicated code.

### CODE REVIEW

**Comment by ColdIce1605** [comment]
can I have a name check as im not sure if this is the right name for the method

**Comment by OverlordsIII** [comment]
> can I have a name check as im not sure if this is the right name for the method

I think the method should probably just be called `createShader`?

File: src/main/java/net/coderbot/iris/gl/program/ProgramBuilder.java [hunk 1]
```diff
@@ -88,4 +76,12 @@ public static ProgramBuilder begin(String name, @Nullable String vertexSource, @
 	public Program build() {
 		return new Program(program, super.buildUniforms());
 	}
+
+	private static GlShader createShaderType(ShaderType ShaderType, String name, @Nullable String source) {
```
Comment (OverlordsIII) [inline src/main/java/net/coderbot/iris/gl/program/ProgramBuilder.java:80]: the parameter `ShaderType` is uppercase and not camel case

**Comment by ColdIce1605** [comment]
> 
> 
> > can I have a name check as im not sure if this is the right name for the method
> 
> I think the method should probably just be called `createShader`?

imma call it buildShader

**Pull request merged**